server/src/middlewares/auth.js

```diff
@@ -0,0 +1,21 @@
+import errorMessage from '../utils/errorMessage';
+
+export const signupValidation = async (req, res, next) => {
+  if (!req.body.email || !req.body.username || !req.body.password) {
+    return errorMessage({
+      status: 400,
+      message: 'Missing some imput fields. Please fill the details.',
+    }, req, res);
+  }
+  return next();
+};
+
+export const signinValidation = async (req, res, next) => {
+  if (!req.body.email || !req.body.password) {
+    return errorMessage({
+      status: 404,
```

Comment:
404 is not a correct code here. 
Your creds are missing, but route/ page is still intact

Reply:
Corrected, thanks